Adds UniFi Protect integration (#20851)
home-assistant/home-assistant.github.io: source/_integrations/unifiprotect.markdown

---
title: UniFi Protect
description: Instructions on how to configure UniFi Protect integration by Ubiquiti.
ha_category:
  - Hub
  - Camera
ha_release: 2022.2
ha_iot_class: Local Push
ha_config_flow: true
ha_quality_scale: platinum
ha_codeowners:
  - '@briis'
  - '@AngellusMortis'
  - '@bdraco'
ha_domain: unifiprotect
ha_platforms:
  - camera
---

The UniFi Protect integration, adds support for retrieving Camera feeds and Sensor data from an [UniFi Protect application](https://ui.com/camera-security) by [Ubiquiti Networks, inc.](https://www.ui.com/) that is running on an UniFi OS Console.

## Prerequisites

### Hardware Support

> This Integration supports all UniFi OS Consoles that can run UniFi Protect. Currently this includes:

Reviewer: ```suggestion
This Integration supports all UniFi OS Consoles that can run UniFi Protect. Currently, this includes:
```

Example: as of `2022.2.0` of Home Assistant, UniFi Protect `1.21.0-beta.1` is the newest UniFi Protect version. So the recommended versions of UniFi Protect to run for a `2022.2.0` version of Home Assistant are `1.20.0`, `1.20.1` or `1.21.0-beta.1`.

### Local User

You will need a local user created in your UniFi OS Console to log in with. Ubiquiti Cloud Users will **not** work.

1. Login to your *Local Portal* on your UniFiOS device, and click on *Users*
1. In the upper right corner, click on *Add User*
1. Click *Add Admin*, and fill out the form. Specific Fields to pay attention to:
    * Role: Must be *Limited Admin*
    * Account Type: *Local Access Only*
    * CONTROLLER PERMISSIONS - Under UniFi Protect, select Administrators. **NOTE**: Other roles may work, but only the default Administrators rule is fully tested.
1. Click *Add* in at the bottom Right.

**HINT**: A few users have reported that they had to restart their UDMP device after creating the local user for it to work. So if you get some kind of *Error 500* when setting up the Integration, try restart the UDMP.

![ADMIN_UNIFIOS](/images/integrations/unifiprotect/unifi_os_admin.png)

### Camera Streams

> The Integration uses the RTSP(S) Streams as the Live Feed source, so this needs to be enabled on each camera to ensure you can stream your camera in Home Assistant. This may already be enabled by default, but it is recommended to just check that this is done. To check and enable the the feature

Reviewer: ```suggestion
The Integration uses the RTSP(S) Streams as the Live Feed source, so this needs to be enabled on each camera to ensure you can stream your camera in Home Assistant. This may already be enabled by default, but it is recommended to just check that this is done. To check and enable the feature:
```

> 1. open UniFi Protect and click on *Devices*

Reviewer: All the other lines have title case except this one

1. Select *Manage* in the Menu bar at the top

> 1. Click on the + Sign next to RTSP

Reviewer: ```suggestion
1. Click on the + sign next to RTSP
```

> 1. Enable minimum 1 stream out of the 3 available. The Stream with the Highest resolution as the default enabled one.

Reviewer: ```suggestion
1. Enable a minimum 1 stream out of the 3 available. The Stream with the Highest resolution is the default enabled one.
```

{% include integrations/config_flow.md %}

## Features

All known UniFi Protect devices are should be supported. Each UniFi Protect device will get a variety of entities added for each of the different entity platforms.

### Cameras

Each UniFi Protect camera will get the following entities added:

* **Camera** - A camera entity for each camera channel and RTSP(S) combination found for each camera (up to 6). Only the highest resolution RTSPS camera entity will be enabled by default.

## Troubleshooting

### Delay in Video Feed

> The default settings on the stream integration, will give you a 5-15+ second delay. You can reduce this delay to 1-3 seconds, by enabling [LL-HLS in the stream integration](/integrations/stream/#ll-hls). You will also want to put an HTTP/2 reserve proxy in front of Home Assistant so you can have connection pooling. If you do not add a reverse proxy, you may start to get "Waiting for Websocket..." messages while trying to view too many camera streams at once. One way to do this is using the official NGINX Proxy Add-on:

Reviewer: ```suggestion
The default settings on the stream integration will give you a 5-15+ second delay. You can reduce this delay to 1-3 seconds, by enabling [LL-HLS in the stream integration](/integrations/stream/#ll-hls). You will also want to put an HTTP/2 reserve proxy in front of Home Assistant so you can have connection pooling. If you do not add a reverse proxy, you may start to get "Waiting for Websocket..." messages while trying to view too many camera streams at once. One way to do this is using the official NGINX Proxy Add-on:
```

> [![NGINX Proxy Add-on](https://my.home-assistant.io/badges/supervisor_addon.svg)](https://my.home-assistant.io/redirect/supervisor_addon/?addon=core_nginx_proxy)

Reviewer 2: Don't use direct my links, instead use

`{% my supervisor_addon addon="core_nginx_proxy" badge %}`

---
title: UniFi Protect
description: Instructions on how to configure UniFi Protect integration by Ubiquiti.
ha_category:
  - Hub
  - Camera
ha_release: 2022.2
ha_iot_class: Local Push
ha_config_flow: true
ha_quality_scale: platinum
ha_codeowners:
  - '@briis'
  - '@AngellusMortis'
  - '@bdraco'
ha_domain: unifiprotect
ha_platforms:
  - camera
---

The UniFi Protect integration, adds support for retrieving Camera feeds and Sensor data from an [UniFi Protect application](https://ui.com/camera-security) by [Ubiquiti Networks, inc.](https://www.ui.com/) that is running on an UniFi OS Console.

## Prerequisites

### Hardware Support

This Integration supports all UniFi OS Consoles that can run UniFi Protect. Currently this includes:

* UniFi Protect Network Video Recorder (**UNVR**)
* UniFi Protect Network Video Recorder Pro (**UNVRPRO**)
* UniFi Dream Machine Pro (**UDMP**)
* UniFi Cloud Key Gen2 Plus (**CKGP**) firmware version v2.0.24+

> Ubiquiti released v2.0.24 as an official firmware release for the CloudKey+, and it is recommended that people upgrade to this UniFi OS based firmware for their CloudKey+, as this gives a much better realtime experience.
> 

Reviewer: ```suggestion
```

I think this can be dropped as its been a while and they can figure this out from UI instead

---
title: UniFi Protect
description: Instructions on how to configure UniFi Protect integration by Ubiquiti.
ha_category:
  - Hub
  - Camera
ha_release: 2022.2
ha_iot_class: Local Push
ha_config_flow: true
ha_quality_scale: platinum
ha_codeowners:
  - '@briis'
  - '@AngellusMortis'
  - '@bdraco'
ha_domain: unifiprotect
ha_platforms:
  - camera
---

The [UniFi Protect Integration](https://ui.com/camera-security) by [Ubiquiti Networks, inc.](https://www.ui.com/), adds support for retrieving Camera feeds and Sensor data from a UniFi Protect installation on either a UniFi OS Console running UniFi Protect.

{% include integrations/config_flow.md %}

## Device Support

All known UniFi Protect devices are should be supported. Each device will get a variety of entities added for each of the different entity platforms.

### Camera

> A camera entity for each camera channel and RTSP(S) combination found on the NVR device will be created. Only the highest resolution RTSPS camera entity will be enabled by default. Changing your RTSP settings in UniFi Protect will require you to reload the `unifiprotect` integration or restart Home Assistant.

Reviewer 2: Single line chapter? With mixed content. It contains something about the camera's RTSP (which isn't a thing an end-user has to deal with?), and contains instructions on restarting. `uinifiprotect` isn't code, so shouldn't be code fenced, and restarting Home Assistant is not needed at all.

Author: This was originally a bullet list on the HACS README. One per entity platform. I was thinking a header made it look a bit better (reference for how the original looked: https://github.com/briis/unifiprotect/)

Reviewer 2: It doesn't make sense on this page IMHO. There is trouble shooting, something out a protocol they shouldn't care about, entities, channels, restarting integrations and Home assistant... all in that single line. That is out of focus and out of place IMHO.

Author: The bit about reloading/restarting was new. I can remove that. But this section is designed to just list all of the things the integration adds. So right now, it just had the single header for the camera platform as that is all that is in the initial PR, but it will be longer. 

If you do not think this section is useful, I can just scrap it all together.

Author: Reworked this so it is a bullet list of added entities per device type (camera, light, sensor, viewport, NVR).

## UniFi Protect Support

### Hardware Support

This Integration supports all UniFi OS Consoles that can run UniFi Protect. Currently this includes:

* UniFi Protect Network Video Recorder (**UNVR**)
* UniFi Protect Network Video Recorder Pro (**UNVRPRO**)
* UniFi Dream Machine Pro (**UDMP**)
* UniFi Cloud Key Gen2 Plus (**CKGP**) firmware version v2.0.24+

Ubiquiti released v2.0.24 as an official firmware release for the CloudKey+, and it is recommended that people upgrade to this UniFi OS based firmware for their CloudKey+, as this gives a much better realtime experience.

CKGP with Firmware v1.x **do NOT run UniFiOS**, you must upgrade to firmware v2.0.24 or newer.

### Software Support

> The absolute **minimal** software version is `1.20.0` for UniFi Protect. If you have an older version, you will get errors trying to set up the integration. However, the general advice is the latest 2 minor versions of UniFi Protect and hardware supported by those are supported. Since UniFi Protect has its own release cycle, you should only upgrade UniFi Protect _after_ the next Home Assistant release comes out to ensure the new version is fully supported.

Reviewer 2: Isn't this a prerequisite? but that chapter is listed below?

Example: as of `2022.2.0` of Home Assistant, UniFi Protect `1.21.0-beta.1` is the newest UniFi Protect version. So the recommended versions of UniFi Protect to run for a `2022.2.0` version of Home Assistant are `1.20.0`, `1.20.1` or `1.21.0-beta.1`.

## Prerequisites

> Before you install this Integration you need to ensure that the following two settings are applied in UniFi Protect.

Reviewer 2: Let's make sure one reads this before setting up the integration... right now, this is listed halfway down the page.

Author: Moved it up to the top. Should it be above the configuration section?

### Local User

1. Login to your *Local Portal* on your UniFiOS device, and click on *Users*
1. In the upper right corner, click on *Add User*
1. Click *Add Admin*, and fill out the form. Specific Fields to pay attention to:
    * Role: Must be *Limited Admin*
    * Account Type: *Local Access Only*
    * CONTROLLER PERMISSIONS - Under UniFi Protect, select Administrators. **NOTE**: Other roles may work, but only the default Administrators rule is fully tested.
1. Click *Add* in at the bottom Right.

**HINT**: A few users have reported that they had to restart their UDMP device after creating the local user for it to work. So if you get some kind of *Error 500* when setting up the Integration, try restart the UDMP.

![ADMIN_UNIFIOS](/images/integrations/unifiprotect/unifi_os_admin.png)

### Camera Streams

> The Integration uses the RTSP(S) Streams as the Live Feed source, so this needs to be enabled on each camera. With the latest versions of UniFi Protect, the stream is enabled per default, but it is recommended to just check that this is done. To check and enable the the feature

Reviewer 2: Again camera streams? That was already listed above?

1. open UniFi Protect and click on *Devices*
2. Select *Manage* in the Menu bar at the top
3. Click on the + Sign next to RTSP
4. Enable minimum 1 stream out of the 3 available. UniFi Protect will select the Stream with the Highest resolution as the default enabled one.

#### Minimizing Latency

To minimize the latency to your UniFi Protect cameras, you will want to [enable LL-HLS in the stream integration](/integrations/stream/#ll-hls). You will also want to put an HTTP/2 reserve proxy in front of Home Assistant so you can have connection pooling. If you do not add a reverse proxy, you may start to get "Waiting for Websocket..." messages while trying to view too many camera streams at once.

##### Resources

* NGINX Proxy Home Assistant OS Add-on: [![NGINX Proxy Add-on](https://my.home-assistant.io/badges/supervisor_addon.svg)](https://my.home-assistant.io/redirect/supervisor_addon/?addon=core_nginx_proxy)

> * [Community Guide for NGINX](https://community.home-assistant.io/t/reverse-proxy-using-nginx/196954)

Reviewer 2: This should not be listed.

Author: This gets asked _a lot_ by people using the current HACS integration. "Why are do my cameras have a 15 second delay?"

Definetely interested in any other suggestions on how to mitigate it, but we need something.

Reviewer 2: We do not support or recommend community guides; thus should not be listed as such.

Author: Oh, so just remove the community guide. That I can do. I can just change it to something like "if you are not using HA OS, look up a guide".

Reviewer 2: Why not explain what it needs instead of saying "look up a guide"?

Author: I mean, I mention you need a HTTP/2 reverse proxy, but I am trying to avoid this being an official guide on how to do that. I mentioned to @allenporter it would be nice if we had an official one since using LL-HLS for the stream integration _strongly_ recommends having HTTP/2, but I think the docs for this integration it is out of scope.

Reviewer 2: >  but I am trying to avoid this being an official guide on how to do that. 

Correct. So don't do that.

Author: Cleaned up the wording and only reference the NGINX add-on making it less guide-like.
